sources/xfscli/src/bin/xfs_admin.rs

```diff
@@ -0,0 +1,211 @@
+use std::ops::Add;
+use std::process::exit;
+use std::process::Command;
+use std::str;
+
+use argh::FromArgs;
+
```

Comment:
Its probably worth linking to the upstream shell script it aims to replace, for instance: https://git.kernel.org/pub/scm/fs/xfs/xfsprogs-dev.git/tree/db/xfs_admin.sh It took me a bit of time to find this and start comparing.